Gems/Atom/Feature/Common/Assets/Shaders/Materials/Skin/Skin_PixelGeometryData.azsli

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+// This #define magic lets you use the PixelGeometryData class in this file without making it the final PixelGeometryData
+// used in your shader. Simply #define PixelGeometryData to your custom definition before including this file
+//
+#ifndef PixelGeometryData
+#define PixelGeometryData       PixelGeometryData_Skin
+#endif
+
+#include "../BasePBR/BasePBR_PixelGeometryData.azsli"
+
+class PixelGeometryData_Skin
```

Comment:
In some places you have examples of using a future class inheritance feature, but not here. I assume that once inheritance is available you're going to update all these structures shown in the PR right?

Reply:
Yep, will do a complete pass on all of these to use class inheritance. I hate duplicate code too much not to 😄